modules/apps/users-admin/users-admin-impl/src/main/java/com/liferay/users/admin/internal/search/spi/model/query/contributor/UserKeywordQueryContributor.java

```diff
@@ -81,7 +81,7 @@ public void contribute(
 					_getTrailingWildcardQuery("emailAddressDomain", keywords),
 					BooleanClauseOccur.SHOULD);
 				booleanQuery.add(
-					_getTrailingWildcardQuery("screenName", keywords),
+					_getTrailingWildcardQuery("screenName.text", keywords),
```

Comment:
Do we need this? Do we need this for emailAddress and emailAddressDomain either, because the KeywordFieldQueryBuilder is adding a trailing wildcard query for those field anyways?

Reply:
I'll take a look

Reply:
The affect of this last 3 contributions for "emailAddress", "emailAddressDomain", and "screenName.text" are these clauses (inside a large should):
```
                    {
                      "wildcard": {
                        "emailAddress": {
                          "wildcard": "tes*"
                        }
                      }
                    },
                    {
                      "wildcard": {
                        "emailAddressDomain": {
                          "wildcard": "tes*"
                        }
                      }
                    },
                    {
                      "wildcard": {
                        "screenName.text": {
                          "wildcard": "tes*"
                        }
                      }
                    }
```
"screenName" (not "screenName.text") does already have a contribution earlier that looks like this:
```
                    {
                      "bool": {
                        "must": [
                          {
                            "wildcard": {
                              "screenName": {
                                "wildcard": "tes*"
                              }
                            }
                          }
                        ],
                        "should": [
                          {
                            "term": {
                              "screenName": {
                                "value": "tes"
                              }
                            }
                          }
                        ]
                      }
                    }
```

Reply:
So the question when it comes to screenName is how will the wildcard queries function differently between text fields and keyword fields. I did find an example of a difference: when a user is created with screenName "firetruck-red", the keyword screenName field clause `"wildcard": "red*"` won't match, where the screenName.text field clause will match. If that difference is worth persisting, then we should keep the second wildcard clause.

Reply:
As for the "emailAddress" and "emailAddressDomain" fields, these aren't contributed elsewhere and they aren't controlled by the KeywordFieldQueryBuilder which is [configured just with these fields](https://github.com/liferay/liferay-portal/blob/c0b0d71d74d50d78fa1c200eace1ca01e04ae18b/modules/apps/portal-search/portal-search-api/src/main/java/com/liferay/portal/search/configuration/QueryPreProcessConfiguration.java#L43).

Reply:
So in my opinion, we should leave the current code as is, let me know if you agree @peerkar.

Reply:
Sounds good, thanks for checking!